cms/djangoapps/contentstore/middleware.py

```diff
@@ -0,0 +1,77 @@
+"""
+Middlewares for contentstore app.
+"""
+
+import logging
+
+from django.core.exceptions import PermissionDenied
+from django.urls import Resolver404, resolve
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import AssetKey, CourseKey, UsageKey
+
+from common.djangoapps.student.roles import CourseLimitedStaffRole
+
+logger = logging.getLogger(__name__)
+
+
+class LimitedStaffAccessMiddleware:
+    """
+    Middleware to check access for limited staff users.
+    """
+
+    WHITELISTED_PAGES = ("/static",)
```

Comment:
Technically, this is all configurable, and may change at some point for a given deployment. But again, we have the resolver's match object, so we can specifically check if `match.func == django.views.static.serve`.

That being said, I don't think that the actual serving of static assets of this nature (the JS/CSS/images we generate in code) is going to be a potential issue. It's all public anyhow, and it's going to be served by a proxy of some sort or another in production instances, at which point it doesn't go through Django at all. What I was worried about were files uploaded by the course teams, which is served out of the `StaticContentServer` middleware. Basically, I think that as long as this new middleware comes *after* `StaticContentServer`, we should be okay.